runtime/makelib/mkconstants.mk.ftl

```diff
@@ -24,7 +24,10 @@
 </#list>
 
 # Define the Java Version we are compiling
+# Ensure VERSION_MAJOR hence JAVA_SPEC_VERSION not empty
+ifndef VERSION_MAJOR
 VERSION_MAJOR:=9
```

Comment:
This can be more simply expressed as `VERSION_MAJOR ?= 9`; there's no need for `ifndef`.

Reply:
Updated with comments.